src/system-fetch.js

```diff
@@ -59,6 +59,29 @@
       }
     };
   }
+  else if (typeof fetch != 'undefined') {
```

Comment:
`fetch` isn't actually specific to service workers but the "new" [promise-based request API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) that is intended to replace XMLHttpRequest for browser windows and web workers. Service worker are special workers in that case as they explicit do not allow the older XHR methods.

Maybe one could test for `self.fetch` as `self` is the `Window`, `WorkerGlobalScope` or `ServiceWorkerGlobalScope` object at least in a modern Chrome. It is still nothing that cannot be overridden.

If using `fetch` should be restricted to only service workers (even as it works in windows and web workers) something like this may work ([source](https://stackoverflow.com/a/18002694)):

```
if (typeof ServiceWorkerGlobalScope !== 'undefined' && self instanceof ServiceWorkerGlobalScope)
```